pkg/csi/service/common/util.go

```diff
@@ -436,3 +437,37 @@ func GetCSINamespace() string {
 	}
 	return CSINamespace
 }
+
+func GetValidatedCNSVolumeInfoPatch(ctx context.Context, cnsSnapshotInfo *cnsvolume.CnsSnapshotInfo, cnsVolumeInfo *cnsvolumeinfov1alpha1.CNSVolumeInfo) map[string]interface{} {
+	log := logger.GetLogger(ctx)
+	var patch map[string]interface{}
+	if cnsSnapshotInfo.AggregatedSnapshotCapacityInMb == -1 {
+		log.Infof("Couldn't retrieve aggregated snapshot capacity for volume %d and snapshot %d",
+			cnsVolumeInfo.Spec.VolumeID, cnsSnapshotInfo.SnapshotID)
+		patchAnnotation := MergeMaps(cnsVolumeInfo.Annotations,
+			map[string]string{"csi.vsphere.missing-snapshot-aggregated-capacity": "true"})
+		patch = map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": patchAnnotation,
+			},
+			"spec": map[string]interface{}{
+				"validaggregatedsnapshotsize": false,
+			},
+		}
+	} else {
+		log.Infof("retrived aggregated snapshot capacity for volume %d and snapshot %d",
+			cnsVolumeInfo.Spec.VolumeID, cnsSnapshotInfo.SnapshotID)
+		delete(cnsVolumeInfo.Annotations, "csi.vsphere.missing-snapshot-aggregated-capacity")
```

Comment:
If this annotation doesn't exist, then there is no impact since we are not considering return status of delete, right?

Reply:
no there will be no impact of this,  it is safe to delete map entries like this